### PR TITLE
implement camera flight controls

### PR DESCRIPTION
**Pull request by PTLux1517**
When the mouse cursor is in the viz area you can fly using:
```
Right-handed mouse:
"w" -> forward (or mouse wheel)
"a" -> left
"s" -> backward (or mouse wheel)
"d" -> right
"q" -> down
"e" -> up

Left-handed mouse (with num-pad):
"8" -> forward (or mouse wheel)
"4" -> left
"5" -> backward (or mouse wheel)
"6" -> right
"7" -> down
"9" -> up

Ambi/intuitive:
"🠩" -> forward (or mouse wheel)
"🠨" -> left
"🠫" -> backward (or mouse wheel)
"🠪" -> right
"PageDown" -> down
"PageUp" -> up
```

## As mentioned in Discord, this PR addresses the coordinate system conversion:
One further discovery is that all 3 axes in COBR's coordinate system are different than THREE's:
```
THREE.x -> COBR.-x
THREE.y -> COBR.z
THREE.z -> COBR.y
```
Unfortunately there doesn't appear to be a neat solution for changing THREE's coordinate system (see https://discourse.threejs.org/t/convert-from-one-coordinate-system-to-another/13240), so it seems to me like the most natural approach would be to just do the translation when assigning x,y,z to each marker mesh. That way we can use all of THREE's provided assets like the camera, grid, axis helper, etc in their natural orientation and not have to worry about applying odd rotations to everything.

### CODE REVIEW

**Comment by PTLux1517** [comment]
The axis helper in #98 will show the user COBR's coordinate system, so THREE's will be invisible to them.

**Review by javitrejo14 — APPROVED** [review]

File: src/RenderView.tsx [hunk 1]
```diff
@@ -66,6 +74,59 @@ export default function RenderView(props: Props) {
 		return result;
 	});
 
+	const [cameraControls] = useState(() => new PointerLockControls(camera, renderer.domElement));
+
+	/* Use camera controls */
+	useEffect(() => {
+		const moveMeters = 0.05;
+		/* Only allow flight while mouse is in visualization area */
+		renderer.domElement.addEventListener('mouseenter', () => document.addEventListener("keydown", keyPressHandler, false), false);
```
Comment (vpzomtrrfrt) [inline src/RenderView.tsx:83]: if you add event listeners in useEffect you should remove them in the cleanup function

File: src/RenderView.tsx [hunk 2]
```diff
@@ -66,6 +74,59 @@ export default function RenderView(props: Props) {
...
+		renderer.domElement.addEventListener('mouseleave', () => document.removeEventListener("keydown", keyPressHandler, false), false);
+		/* Flight controls (keys) */
+		const keyPressHandler = (e: KeyboardEvent) => {
+			switch (e.key) {
```
Comment (vpzomtrrfrt) [inline src/RenderView.tsx:87]: I think using `e.code` here would be better since it works with shift & non-qwerty layouts too

**Review by vpzomtrrfrt — CHANGES_REQUESTED** [review]

File: src/RenderView.tsx (hunk 1 above)
Comment (PTLux1517) [reply src/RenderView.tsx]: Done. I'm still getting the hang of React. That makes sense.

File: src/RenderView.tsx (hunk 2 above)
Comment (PTLux1517) [reply src/RenderView.tsx]: Ah, good to know. Also done.

**Review by vpzomtrrfrt — APPROVED** [review]
would be nice to have smoother movement when held but this is a good start

**Pull request merged**